static/js/components/tiles/map_tile.tsx

```diff
@@ -256,175 +286,222 @@ export function getReplacementStrings(
+  const rawDataArray = [];
+  for (const layer of layers) {
+    const place = props.dataSpec
+      ? { name: "", dcid: layer.parentPlace, types: [] }
+      : props.place;
+    const geoJsonParams = {
+      placeDcid: layer.parentPlace,
+      placeType: layer.enclosedPlaceType,
     };
-    return rawToChart(
-      rawData,
-      props.statVarSpec,
-      props.place,
-      props.enclosedPlaceType,
-      props
+    if (props.placeNameProp) {
+      geoJsonParams["placeNameProp"] = props.placeNameProp;
+    }
+    if (props.geoJsonProp) {
+      geoJsonParams["geoJsonProp"] = props.geoJsonProp;
+    }
+    const geoJsonPromise = axios
+      .get(`${props.apiRoot || ""}/api/choropleth/geojson`, {
+        params: geoJsonParams,
+        paramsSerializer: stringifyFn,
+      })
+      .then((resp) => resp.data);
+    const borderGeoJsonPromise = axios
+      .post(`${props.apiRoot || ""}/api/choropleth/node-geojson`, {
+        geoJsonProp: "geoJsonCoordinates",
+        nodes: [layer.parentPlace],
+      })
+      .then((resp) => resp.data);
+    const dataDate =
+      layer.variable.date || getCappedStatVarDate(layer.variable.statVar);
+    const facetIds = layer.variable.facetId ? [layer.variable.facetId] : null;
+    const placeStatPromise: Promise<PointApiResponse> = getPointWithin(
+      props.apiRoot,
+      layer.enclosedPlaceType,
+      layer.parentPlace,
+      [layer.variable.statVar],
+      dataDate,
+      [],
+      facetIds
     );
-  } catch (error) {
-    removeSpinner(props.id);
-    return null;
+    const populationPromise: Promise<SeriesApiResponse> = props.statVarSpec
+      .denom
+      ? getSeriesWithin(
+          props.apiRoot,
+          layer.parentPlace,
+          layer.enclosedPlaceType,
+          [layer.variable.denom]
+        )
+      : Promise.resolve(null);
+    const parentPlacesPromise = props.parentPlaces
+      ? Promise.resolve(props.parentPlaces)
+      : axios
+          .get(
+            `${props.apiRoot || ""}/api/place/parent?dcid=${layer.parentPlace}`
+          )
+          .then((resp) => resp.data);
+    try {
+      const [geoJson, placeStat, population, parentPlaces, borderGeoJsonData] =
+        await Promise.all([
+          geoJsonPromise,
+          placeStatPromise,
+          populationPromise,
+          parentPlacesPromise,
+          borderGeoJsonPromise,
+        ]);
+      // Only draw borders for containing places without 'wall to wall' coverage
+      const borderGeoJson = shouldShowBorder(layer.enclosedPlaceType)
+        ? borderGeoJsonData
+        : undefined;
+      const rawData = {
+        borderGeoJson,
+        enclosedPlaceType: layer.enclosedPlaceType,
+        geoJson,
+        parentPlaces,
+        place,
+        placeStat,
+        population,
+      };
+      rawDataArray.push(rawData);
+    } catch (error) {
+      removeSpinner(props.id);
+      return null;
+    }
   }
+  return rawToChart(rawDataArray, layers[0].variable, props);
```

Comment:
add a comment that for now, all layers will have the same variable

Reply:
Added comment, and a TODO to add multi-variable functionality.